### DISCUSSION TITLE
need advice with direction

### DISCUSSION THREAD

**Discussion by molleweide** [pwntester/octo.nvim#253] [Q&A] [answered]
Hi pwntester and all who are following,

I would like to try to add support for discussions but I would like to ask for some advice before I get
started. Would you mind just pointing out which would be the main functions that I would have to modify
in order to achieve this. 

I am wondewing are these the main files that I would have to modify:

1. `octo.lua` and add a check for `discussion` in all places, and
2. the `graphql.lua` file.

I have browsed through the code a little bit and it seems that these two are the main files, or am I wrong?

**Comment by pwntester** [comment, marked as answer]
Hi @molleweide Thanks for your offer to work on this.

From the top of my head you will need to touch the following files:
- `commands.lua` to add the new actions and commands for discussions such as `octo discusstion create` -> `create` action
- `config.lua` and `mappings.lua` for key bindings
- `octo-buffer.lua` will need to handle a new `kind` for discussion buffers
- you may need new writers in `writers.lua` to render new discussion UI elements such as the accepted response
- all graphql queries should go into `graphql.lua`
- `init.lua` should also account for this new discussion `kind`

If you finally decide to implement support for it, let me know and I will try to help

**Reply by molleweide** [reply]
Great thanks man!!

I would very much like this feature. It might be difficult for me to do this but it will be nice to try!

**Comment by williambdean** [comment]
Closing with discussion supported implemented in https://github.com/pwntester/octo.nvim/pull/918. There are still some features that are coming along. I have them in #672
